JcMinarro/Philology
# Read text resource id from attribute set

For a more complex case when a text is set using a [custom attribute](https://github.com/JcMinarro/Philology/pull/27/files#diff-bda8cde6acf4c1a61ec6e19e0d02204eR113) that is set from a style being applied as an [android:theme](https://github.com/JcMinarro/Philology/pull/27/files#diff-bda8cde6acf4c1a61ec6e19e0d02204eR115) the existing implementation of  `ViewTransformer#reword` does not handle.

Turns out we can cover all the possible cases by simplifying the way we read the text attributes (`"text", "hint", "title", "subtitle"`) from the `AttributeSet`.

Since `ViewTransformer#reword` is only reapplying the text on a `View` by identifying the resourceId of that text, we should read it using Android attributes, over [hardcoded](https://github.com/JcMinarro/Philology/pull/27/files#diff-2210131d9f1ab00ee2a56364c66174d1L9) String attribute names.

### Changes
1. Implemented a new `Context.getStringResourceId` extension that should be used to find the text resource id using a given attribute
2. Adjusted `TextViewTransformer`, `ToolbarTransformer` and `SupportToolbarViewTransformer` to use the new implementation
3. Added translatable `hint`'s for the plural edit texts in the sample application

📓 **Note:** (refers to this [comment](https://github.com/JcMinarro/Philology/pull/26#discussion_r355895627))
Considering the changes made in the current PR the [ViewTransformer#setTextIfExists](https://github.com/JcMinarro/Philology/pull/27/files#diff-c14d365e545a0c9dc45644fb32d4ac1eL10) is not needed anymore and [ViewTransformer#reword](https://github.com/JcMinarro/Philology/pull/27/files#diff-c14d365e545a0c9dc45644fb32d4ac1eR10) will remain the only public function to be used with the help of [Context.getStringResourceId](https://github.com/JcMinarro/Philology/pull/27/files#diff-c14d365e545a0c9dc45644fb32d4ac1eR13) which simplifies the read of string resources id for the required attribute.


### How to test
1. Run the sample app in `Russian` or `Spanish`
2. Make sure all text views are translated including hints
